### ISSUE TITLE
--region not handling sts endpoint as expected; still needs AWS_REGION environment variable

### ISSUE THREAD

**Issue by p-kossa** [uswitch/kiam#344]
I've been testing kiam in `us-gov-west-1` while using the `--region=us-gov-west-1` flag in the server DaemonSet manifest but I kept seeing this error:

`error retrieving credentials: InvalidClientTokenId: The security token included in the request is invalid`

when deploying an AWS CLI test pod to assume the pod role. Upon searching the closed issues here I came across this https://github.com/uswitch/kiam/issues/278 experiencing the same issue I was despite using the flag as intended.

Once I added `AWS_REGION` env var to the kiam server DaemonSet the pod was able to assume role successfully. 

It's my understanding that the `--region` option should handle any region-specific functions in kiam, is the requirement for the `AWS_REGION` env var intentional or could this be a bug?

**Comment by darkdragn** [comment]
I'm thinking that this isn't the fault of their endpoint resolution. I'm thinking it's a side effect of the base AWS package for golang. I'm only bringing it up because I ran into the same issue for Vault's AWS Auth module, and even though I expressly set the region in the Auth config, it still didn't work unless I also set AWS_REGION.

I'm thinking that the endpoint resolution is needed for the DNS target, but AWS_REGION is being used for header signing on the requests or something to that effect.
